### DISCUSSION TITLE
Sites per tenant isolated

### DISCUSSION THREAD

**Discussion by Alik2015** [OrchardCMS/OrchardCore#14251] [General]
I am trying to host multiple decoupled sites in a site-per-tenant scenario.

I created 2 modules using razor pages and if I go to module1 I see my page and same with /module2.  My issue is that I find the _layout.cshtml is shared by both modules so both see the same layout file.

How can I isolate the sites so each tenant only sees there own site?

I have been searching for days and can't find a single example for doing this and asking same question on StackOverflow did not get any answers.

I want to manage a single instance or orchasecore and just add/update sites as require per new tenants.

Any advice appreciated.

**Comment by MikeAlhayek** [comment]
Typically we create a default site in OC using the “Software as a Service” recipe. This will allow you to create other tenants using the UI.

**Reply by Alik2015** [reply]
I understand. My issue is all tenants get same layout pulled from base site.

**Reply by MikeAlhayek** [reply]
Make sure your web project has no views. Also, you should be able to assign different themes for different tenants. If all tenants have the same theme name, all will use the same layout

**Reply by Alik2015** [reply]
If I set up project as a theme which is razor page project then when using my theme project I get layout Shape not found.

Project layout
Cms project
Theme project
   Pages
      Index.cshml
      Shared / Layout.cshtml
      Startup.cs

**Comment by ns8482e** [comment]
As I understand in decoupled you don't used orchard's themes and  control the theming yourself by defining your own _Layout.

All modules are available to all sites - if your modules has _Layout then only enable module for one site and create another module for another site.

**Reply by Alik2015** [reply]
> Thanks will try that.
